core/src/main/java/io/undertow/util/Cookies.java

```diff
@@ -417,6 +419,79 @@ private static String unescapeDoubleQuotes(final String value) {
         return new String(tmp, 0, dest);
     }
 
+    private static final String CRUMB_SEPARATOR = "; ";
+
+    /**
+     * Cookie headers form: https://www.rfc-editor.org/rfc/rfc6265#section-4.2.1
+     * If more than one header entry exist for "Cookie", it will be assembled into one that conforms to rfc.
+     * @param headerMap
+     * @return
+     */
+    public static void assembleCrumbs(final HeaderMap headerMap) {
+        final HeaderValues cookieValues = headerMap.get(Headers.COOKIE);
+
+        if (cookieValues != null && cookieValues.size() > 1) {
+            final StringBuilder oreos = new StringBuilder();
+            final String[] _cookieValues = cookieValues.toArray();
+            int slices = _cookieValues.length;
+            for (final String slice : _cookieValues) {
+                oreos.append(slice);
+                slices--;
+                if (slices >= 1) {
+                    oreos.append(CRUMB_SEPARATOR);
+                }
+            }
+            cookieValues.clear();
+            cookieValues.add(oreos.toString());
+        }
+    }
+
+    /**
+     * IF there is single entry that follows RFC separation rules, it will be turned into singular fields. This should be only
+     * used PRIOR to compression.
+     *
+     * @param headerMap
+     */
+    public static void disperseCrumbs(final HeaderMap headerMap) {
```

Comment:
Is this a utility method available to Undertow client users? I ask because I don't see any point invoking this method

Reply:
I did not find example, but RFC wording suggests parts of headers and content can be compressed. If user implements such handler, it can use this method - at least this is how I imagine this pieces should work in undertow?

Reply:
Yes, this is how I understood it as well (I was just double checking the intention of the method). Let's just make this clear in the method javadoc in the final version of the PR